server/controllers/answer.js

```diff
@@ -1,152 +1,221 @@
+            }
+            client.query(answerQueries.getAnswersByQId(questionId))
+              .then((answerResponse) => {
+                const answers = answerResponse.rows;
+                if (answers.find(ans => ans.userid === userId)) {
+                  return errors.unauthorized(res);
+                }
+                client.query(answerQueries
+                  .postAnswer(userId, questionId, answer))
+                  .then((postResponse) => {
+                    client.release();
+                    const [newAnswer] = postResponse.rows;
+                    newAnswer.upvotes = 0;
+                    newAnswer.downvotes = 0;
+
+                    return res.status(201).json({
+                      status: 'success',
+                      message: 'Your answer has been posted',
+                      newAnswer,
+                    });
+                  });
+              });
+          });
+      });
   }
 
   /**
-   * Marks an answer as accepted
+   * Updates an answer
    * @param {object} req Request
    * @param {object} res Response
-   * @returns {object} Accepted answer object or error object if answer is not found
+   * @returns {object} Updated answer object
+   * or error object if answer is not found
    */
-  static acceptAnswer(req, res) {
+  static updateAnswer(req, res) {
     const { questionId, answerId } = req.params;
-    const { userId } = req.body;
-
-    const user = userRepo.getUser(userId);
-    if (!user) {
-      return errors.notFound(res, 'user');
-    }
-
-    const answer = answerRepo.getAnswer(answerId, questionId);
-    if (!answer) {
-      return errors.notFound(res, 'answer');
-    }
-
-    const question = repo.getQuestion(questionId);
-    if (!question) {
-      return errors.notFound(res, 'question');
-    }
-
-    if (question.user.id !== userId) {
-      return errors.unauthorized(res);
-    }
-
-    let acceptedAnswer = question.answers.find(ans => ans.isAccepted);
-    if (acceptedAnswer) {
-      return errors.unauthorized(res);
-    }
+    const { userId, answer } = req.body;
 
-    acceptedAnswer = answerRepo.acceptAnswer(answerId);
-    return res.status(200).json({
-      status: 'success',
-      message: 'your have accepted this answer',
-      acceptedAnswer,
-    });
+    pool.connect()
+      .then((client) => {
+        client.release();
+        client.query(questionQueries.getQuestion(questionId))
+          .then((response) => {
+            const [questionObj] = response.rows;
+            if (!questionObj) {
+              return errors.notFound(res, 'question');
+            }
+            client.query(answerQueries.getAnswer(answerId))
+              .then((ansResponse) => {
+                const [existingAnswer] = ansResponse.rows;
+                if (!existingAnswer) {
+                  return errors.notFound(res, 'answer');
+                }
+                if (existingAnswer.questionid !== questionId) {
```

Comment:
- variable should be named in camelCas. questionid should be questionId
- 404 status code is misused here. If the answer exists (then definitely not 404) but belongs to another question, then return appropriate status code to depict this kind of conflict
_

Reply:
The variable name is that way here because sql somehow modifies camel casing to all lowercase